ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java

```diff
@@ -61,38 +63,42 @@ public void processHead(UInt64 nodeSlot) {
   }
 
   private void processHead(Optional<UInt64> nodeSlot) {
-    withLock(
-            () -> {
-              final Checkpoint finalizedCheckpoint =
-                  recentChainData.getStore().getFinalizedCheckpoint();
-              final Checkpoint justifiedCheckpoint =
-                  recentChainData.getStore().getJustifiedCheckpoint();
-              return recentChainData
-                  .retrieveCheckpointState(justifiedCheckpoint)
-                  .thenCompose(
-                      justifiedCheckpointState -> {
-                        final StoreTransaction transaction =
-                            recentChainData.startStoreTransaction();
-                        final ForkChoiceStrategy forkChoiceStrategy = getForkChoiceStrategy();
-                        Bytes32 headBlockRoot =
-                            forkChoiceStrategy.findHead(
-                                transaction,
-                                finalizedCheckpoint,
-                                justifiedCheckpoint,
-                                justifiedCheckpointState.orElseThrow());
-
-                        recentChainData.updateHead(
-                            headBlockRoot,
-                            nodeSlot.orElse(
-                                forkChoiceStrategy
-                                    .blockSlot(headBlockRoot)
-                                    .orElseThrow(
-                                        () ->
-                                            new IllegalStateException(
-                                                "Unable to retrieve the slot of fork choice head"))));
-                        return transaction.commit();
-                      });
-            })
+    final Checkpoint retrievedJustifiedCheckpoint =
+        recentChainData.getStore().getJustifiedCheckpoint();
+    recentChainData
+        .retrieveCheckpointState(retrievedJustifiedCheckpoint)
+        .thenCompose(
+            justifiedCheckpointState ->
+                withLock(
+                    () -> {
+                      final Checkpoint finalizedCheckpoint =
+                          recentChainData.getStore().getFinalizedCheckpoint();
+                      final Checkpoint justifiedCheckpoint =
+                          recentChainData.getStore().getJustifiedCheckpoint();
+                      if (!justifiedCheckpoint.equals(retrievedJustifiedCheckpoint)) {
+                        LOG.info(
+                            "Skipping head block update as justified checkpoint was updated while loading checkpoint state.");
```

Comment:
if this is going to be an info, does it need to have more context?

Reply:
Added some extra info.